src/Adapter/Alias/Repository/AliasRepository.php

```diff
@@ -0,0 +1,147 @@
+
+use Alias;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Driver\Exception;
+use PrestaShop\PrestaShop\Adapter\Alias\Validate\AliasValidator;
+use PrestaShop\PrestaShop\Core\Domain\Alias\Exception\AliasNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Alias\Exception\CannotAddAliasException;
+use PrestaShop\PrestaShop\Core\Domain\Alias\ValueObject\AliasId;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+use PrestaShop\PrestaShop\Core\Repository\AbstractObjectModelRepository;
+
+class AliasRepository extends AbstractObjectModelRepository
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var string
+     */
+    private $dbPrefix;
+
+    /**
+     * @var AliasValidator
+     */
+    private $aliasValidator;
+
+    public function __construct(
+        Connection $connection,
+        string $dbPrefix,
+        AliasValidator $aliasValidator
+    ) {
+        $this->aliasValidator = $aliasValidator;
+        $this->connection = $connection;
+        $this->dbPrefix = $dbPrefix;
+    }
+
+    /**
+     * Creates new Alias entity and saves to the database
+     *
+     * @param string $searchTerm
+     * @param string[] $aliases
+     *
+     * @return AliasId[]
+     *
+     * @throws CoreException
+     */
+    public function create(string $searchTerm, array $aliases): array
+    {
+        $aliasIds = [];
+
+        foreach ($aliases as $searchAlias) {
+            // As search term is not a primary key, we need make sure that alias and search combination does not exist
+            if ($this->aliasExists($searchAlias, $searchTerm)) {
+                continue;
+            }
+
+            $alias = new Alias();
+            $alias->search = $searchTerm;
+            $alias->alias = $searchAlias;
+            $alias->active = true;
+            $this->aliasValidator->validate($alias);
+
+            $this->addObjectModel($alias, CannotAddAliasException::class);
+
+            $aliasIds[] = new AliasId((int) $alias->id);
+        }
+
+        return $aliasIds;
+    }
+
+    public function get(AliasId $aliasId): Alias
+    {
+        /** @var Alias $alias */
+        $alias = $this->getObjectModel(
+            $aliasId->getValue(),
+            Alias::class,
+            AliasNotFoundException::class
+        );
+
+        return $alias;
+    }
+
+    public function aliasExists(string $alias, string $searchTerm): bool
+    {
+        $qb = $this->connection->createQueryBuilder()
+            ->addSelect('a.id_alias')
+            ->from($this->dbPrefix . 'alias', 'a')
+            ->where('a.search = :search')
+            ->andWhere('a.alias = :alias')
+            ->setParameter('search', $searchTerm)
+            ->setParameter('alias', $alias)
+        ;
+
+        return (bool) $qb->execute()->fetchOne();
+    }
+
+    /**
+     * @param string $searchTerm
+     *
+     * @return string[]
+     *
+     * @throws Exception
+     * @throws \Doctrine\DBAL\Exception
```

Comment:
when method doesn't throw exception directly, then these shouldn't be added in php doc IMO, because it will get outdated pretty fast if used classes doesn't throw the exception anymore